ServerCore/Areas/Identity/UserAuthorizationPolicy/AuthorizationHelper.cs

```diff
@@ -96,6 +96,8 @@ public static async Task IsPuzzleAuthorCheck(AuthorizationHandlerContext authCon
             {
                 authContext.Succeed(requirement);
             }
+
+            dbContext.Entry(puzzle).State = EntityState.Detached;
```

Comment:
What is the goal of this line? It seems like it would only run if the auth check failed - I can't figure out what it's supposed to do

Reply:
The "Succeed" line above is not a return - so this line actually runs in all cases. This is something I discovered this evening, where auth prevents a puzzle from being edited because it is keeping an entity attached.

Reply:
What does the line do then? I thought the only purpose of the call was to set succeeded / failed, which doesn't seem related to this line? Is it detaching a puzzle, and if so what does that mean and why?

Reply:
The auth check is keeping some sort of lock on the puzzle that prevents it from being edited. This line removes that lock. As to why we don't have similar locking issues all over the codebase, I can't answer that. I only know that without this line, I can't edit a puzzle, and with this line, I can.